src/core/componentSystems/PhysicsSystem.js

```diff
@@ -290,49 +219,65 @@ export class PhysicsSystem extends MRSystem {
      * @param {MREntity} entity - the entity being updated
      */
     updateBody(entity) {
-        if (entity.physics.type == 'none') {
+        if (!entity.physics.body) {
             return;
         }
-        entity.object3D.getWorldPosition(this.tempWorldPosition);
+
+        if(entity.compStyle.visibility == 'hidden' && entity.physics.body.isEnabled()) {
+           entity.physics.body.setEnabled(false)
+        } else if (!entity.physics.body.isEnabled()) {
+           entity.physics.body.setEnabled(false)
+        }
+
+        if (entity instanceof MRPanel) {
+            entity.panel.getWorldPosition(this.tempWorldPosition);
+        } else {
+            entity.object3D.getWorldPosition(this.tempWorldPosition);
+        }
         entity.physics.body.setTranslation({ ...this.tempWorldPosition }, true);
 
         entity.object3D.getWorldQuaternion(this.tempWorldQuaternion);
         entity.physics.body.setRotation(this.tempWorldQuaternion, true);
 
-        entity.updatePhysicsData();
-        this.updateCollider(entity);
+        if(entity instanceof MRModel) {
+            this.updateSimpleBody(entity);
+        } else if (entity instanceof MRDivEntity) {
+            this.updateUIBody(entity);
+        }
```

Comment:
this is the same commentary as before:

we have the open issue of MRModel possibly being moved to inherit from MRDivEntity instead of MREntity - make sure this reflects that or has a todo comment to warn for when we make that change